lib/Optimizer/Builder/Intrinsics.cpp

```diff
@@ -536,6 +536,7 @@ static constexpr IntrinsicCode intrinsicTable[] = {
   func.func private @__quantum__qis__convert_array_to_stdvector(!qir_array) -> !qir_array
   func.func private @__quantum__qis__free_converted_stdvector(!qir_array)
 
+  func.func private @__quantum__qis__save_state()
```

Comment:
Does this really require QIR in the loop? It is only available in simulation, no? It would be clearer, if it is not a QIS function, that it will never be a part of a hardware profile and, as a bonus, the QIR verifier will fail to let it slip past as something valid for hardware.